### DISCUSSION TITLE
ISR Support

### DISCUSSION THREAD

**Discussion by jdgamble555** [analogjs/analog#1179] [Ideas]
### Which scope/s are relevant/related to the feature request?

Don't known / other

### Information

I don't really care about ISR support specifically, but automating the `Cache-Control` header in serverless functions.  

I want something that uses the `Cache-Control` header, and I don't have to deal with it.

NextJS used to use [ISR](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props), but now they realize the important part of it is just automatically dealing with caching period. This would be huge for this framework considering NextJS is the only Framework that has it for the moment (SvelteKit confirmed eventually, and Nuxt3 eventually).

I am not a fan of using `fetch` like [Next13](https://nextjs.org/blog/next-13#data-fetching) uses, as we may have our own fetch functions we can't control like with Supabase or Firebase.

Just dealing with the caching is what is important. 

### Describe any alternatives/workarounds you're currently using

We could use [ngx-isr](https://github.com/eneajaho/ngx-isr), but it really uses in-memory cache under-the-hood.

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [X] No

**Comment by nartc** [comment]
cc @eneajaho

**Comment by brandonroberts** [comment]
That's an option

Nitro, the underlying server being used has support for custom route rules including headers for cache control

https://nitro.unjs.io/config#routerules
